### PR TITLE
[#50175] Restrict global price group chart string changes

### PR DESCRIPTION
**Pull request by jdzak**
This is work in progress PR just to get comments.

https://pm.tablexi.com/issues/50175

### CODE REVIEW

**Comment by jdzak** [comment]
Alright, I cleaned this up. It should be ready for review now.

File: app/controllers/concerns/price_group_members_controller.rb [hunk 1]
```diff
@@ -56,4 +57,8 @@ def load_price_group_and_ability!
     @price_group = current_facility.price_groups.find(params[:price_group_id])
     @price_group_ability = Ability.new(current_user, @price_group, self)
   end
+
+  def authorize_price_group_member
+    @price_group_ability.authorize! :create, PriceGroupMember
```
Comment (jhanggi) [inline app/controllers/concerns/price_group_members_controller.rb]: While it's likely fine in this scenario, you're calling this before filter on `destroy` as well, which is technically a different permission. Could you use `action_name`?

File: spec/controllers/account_price_group_members_controller_spec.rb [hunk 2]
```diff
@@ -3,30 +3,64 @@
...
+      end
+    end
+  end
 
   context "new" do
     before :each do
       @method=:get
       @action=:new
-      @params={ :facility_id => @authable.url_name, :price_group_id => @price_group.id }
+      @account=create_nufs_account_with_owner
```
Comment (jhanggi) [inline spec/controllers/account_price_group_members_controller_spec.rb]: Can we clean up some of this formatting and add spaces around the `=`? I'd rather change the whole file to match our organizational styles than continue to adhere to older stylings for consistency. (But I'd only worry about the files that you're currently touching).

File: spec/controllers/account_price_group_members_controller_spec.rb [hunk 3]
```diff
@@ -38,20 +72,23 @@
     before :each do
       @method=:post
       @action=:create
-      account=create_nufs_account_with_owner
-      @params={ :facility_id => @authable.url_name, :price_group_id => @price_group.id, :account_id => account.id }
+      @account=create_nufs_account_with_owner
+      @params={ :facility_id => @authable.url_name, :price_group_id => price_group.id, :account_id => @account.id }
```
Comment (jhanggi) [inline spec/controllers/account_price_group_members_controller_spec.rb]: Can we switch to 1.9 syntax?

**Comment by jhanggi** [comment]
Overall, looks pretty good. I do have a question:

Should a director or whoever be able to click in and see the users/payment sources within the cancer center? Currently, they're able to see the list of CC payment sources, but not the list of users. Should we restrict the show view as well?

File: app/controllers/concerns/price_group_members_controller.rb (hunk 1 above)
Comment (jdzak) [inline app/controllers/concerns/price_group_members_controller.rb]: Whoops, I'll fix that.

File: spec/controllers/account_price_group_members_controller_spec.rb (hunk 2 above)
Comment (jdzak) [inline spec/controllers/account_price_group_members_controller_spec.rb]: Sure.

File: spec/controllers/account_price_group_members_controller_spec.rb (hunk 3 above)
Comment (jdzak) [inline spec/controllers/account_price_group_members_controller_spec.rb]: Yeppers.

**Comment by jdzak** [comment]
Ready for re-review.

**Comment by jdzak** [comment]
Also, regarding your question... I'm double checking with Todd but I believe there were scenarios where it was useful to allow admin, facility admins, and directors to see the payment sources for any price group.

**Comment by jdzak** [comment]
Straight from the horses mouth (Todd is the horse)...

> Yes, any role with rights to the Price group view now should be able to
> view-only the list of Cancer Center users. Likewise, for the other global
> price group, they should be able to view-only the payment sources.
>
> For local price groups, they should be able to edit/view all.

**Comment by jhanggi** [comment]
:+1: 

I want to wait until we get a stable release of the relay pieces before we merge this down.

**Pull request closed without merging**